### PR TITLE
[MLGO] Add ability to extract IR from bazel using aquery

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adds in support for extracting IR from binaries built with bazel through querying the linker command line using bazel aquery.

### CODE REVIEW

File: llvm/utils/mlgo-utils/mlgo/corpus/extract_ir.py [hunk 1]
```diff
@@ -45,8 +45,8 @@ def parse_args_and_run():
     parser.add_argument(
         "--input_type",
         type=str,
-        help="Input file type - JSON, LLD params, or directory.",
-        choices=["json", "params", "directory"],
+        help="Input file type - JSON, LLD params, directory, or bazel aquery.",
+        choices=["json", "params", "directory", "aquery"],
```
Comment (mtrofin) [inline llvm/utils/mlgo-utils/mlgo/corpus/extract_ir.py:49]: nit: `bazel_aquery`?

File: llvm/utils/mlgo-utils/mlgo/corpus/extract_ir_lib.py [hunk 2]
```diff
@@ -316,6 +316,27 @@ def make_spec(obj_file: str):
     return [make_spec(path) for path in paths]
 
 
+def load_bazel_aquery(aquery_json, obj_base_dir: str, output_dir: str):
+    """Creates an object file array by looking at the JSON output of bazel aquery.
+
+    Args:
+      aquery_json: The JSON-formatted output of the bazel aquery command for
```
Comment (mtrofin) [inline llvm/utils/mlgo-utils/mlgo/corpus/extract_ir_lib.py:323]: Is there a link we can put here for what that format is?

**Review by mtrofin — APPROVED** [review]
lgtm, some nits

File: llvm/utils/mlgo-utils/mlgo/corpus/extract_ir_lib.py (hunk 2 above)
Comment (boomanaiden154) [reply llvm/utils/mlgo-utils/mlgo/corpus/extract_ir_lib.py:323]: It's a JSON version of [this proto](https://github.com/bazelbuild/bazel/blob/3f93d377d036d773fd505a18e084425a00fb94ea/src/main/protobuf/analysis_v2.proto#L25). I've added a comment.

**Pull request merged**